homeassistant/components/climate/__init__.py

```diff
@@ -96,20 +62,20 @@
         vol.Inclusive(ATTR_TARGET_TEMP_HIGH, 'temperature'): vol.Coerce(float),
         vol.Inclusive(ATTR_TARGET_TEMP_LOW, 'temperature'): vol.Coerce(float),
         vol.Optional(ATTR_ENTITY_ID): cv.comp_entity_ids,
-        vol.Optional(ATTR_OPERATION_MODE): cv.string,
+        vol.Optional(ATTR_HVAC_MODE): vol.In(HVAC_MODES),
     }
 ))
 SET_FAN_MODE_SCHEMA = vol.Schema({
     vol.Optional(ATTR_ENTITY_ID): cv.comp_entity_ids,
     vol.Required(ATTR_FAN_MODE): cv.string,
 })
-SET_HOLD_MODE_SCHEMA = vol.Schema({
+SET_PRESET_MODE_SCHEMA = vol.Schema({
     vol.Optional(ATTR_ENTITY_ID): cv.comp_entity_ids,
-    vol.Required(ATTR_HOLD_MODE): cv.string,
+    vol.Required(ATTR_PRESET_MODE): cv.string,
 })
-SET_OPERATION_MODE_SCHEMA = vol.Schema({
+SET_HVAC_MODE_SCHEMA = vol.Schema({
     vol.Optional(ATTR_ENTITY_ID): cv.comp_entity_ids,
-    vol.Required(ATTR_OPERATION_MODE): cv.string,
+    vol.Required(ATTR_HVAC_MODE): vol.In(HVAC_MODES),
```

Comment:
This is genius. Finally a way to stop people from trying to work around our pre-defined constants.